components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/NetworkMessageEncoder.cs

```diff
@@ -393,5 +404,42 @@ public class NetworkMessageEncoder : IMessageEncoder {
                 }
             }
         }
+
+        /// <summary>
+        ///  Reads to identifier to show for notification in payload of IoT Hub method
+        ///  Prio 1: DataSetFieldId (need to be read from message)
+        ///  Prio 2: DisplayName - nothing to do, because notification.Id already contains DisplayName
+        ///  Prio 3: ExpandedNodeId
+        /// </summary>
+        /// <param name="notification">Notification, were ID need to be looked up for</param>
+        /// <param name="message">subscription notification message, containing notifications</param>
+        /// <param name="context">service context</param>
+        /// <returns>identifier of payload element</returns>
+        private string GetPayloadIdentifier(MonitoredItemNotificationModel notification, DataSetMessageModel message, ServiceMessageContext context) {
```

Comment:
we do not do those in our code, but seems like you do

Reply:
Different approach, for me this an encoder task